pytorch_lightning/callbacks/base.py

```diff
@@ -27,6 +27,10 @@ class Callback(abc.ABC):
     Subclass this class and override any of the relevant hooks
     """
 
+    def on_before_accelerator_backend_setup(self, trainer, pl_module):
```

Comment:
do we need to add a new callback hook? can't it be done with the existing ones?

Reply:
Great question ! 

We have `on_init_start, on_init_end` in __init__ function. We don't have access to the model.

Then we have on_fit_start in fit function, but `configure_optimizers` had already being called during accelerator_backend.setup().

Therefore, I needed to introduce a new callback, so I could freeze the model before `configure_optimizers` call and then filter parameters if they don't require gradients with
`optimizer = optim.Adam(filter(lambda p: p.requires_grad, self.parameters()), lr=self.lr)`

Hope it answer your question.

Reply:
In that case, I think `on_before_configure_optimizers` would be a better name. Easier for users to understand.

Can you also mention the added hook in the CHANGELOG?